File: salimmo-parent/property-service/src/main/java/org/propertyservice/dto/InnerPropertyDto.java
package org.propertyservice.dto;

import lombok.*;


@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString
public class InnerPropertyDto {
    private Long id;
    //@JsonIgnoreProperties("innerProperty")
    private PropertyDto property;
    private int nbrOfLevels;
    private int nbrOfPieces;
    private int nbrOfRooms;
    private int nbrOfRoomsGroundFloor;
    private int nbrOfParentalRooms;
    private int nbrOfShowerRooms;
    private int nbrOfShowerRoomsGroundFloor;
    private int nbrOfBathrooms;
    private int nbrOfToilets;
    private int nbrOfToiletsGroundFloor;
    private String kitchenType;
    private Boolean existingBasement;
    private Boolean existingWashRoom;
    private Boolean existingCave;
    private int nbrOfCaves;
    private Boolean existingCellar;
}

File: salimmo-frontend/src/app/models/innerproperty/innerPropertyDto.ts
import {PropertyDto} from "../property/propertyDto";

export class InnerPropertyDto {
  id!:number;
  property!:PropertyDto;
  nbrOfLevels!:number;
  nbrOfPieces!:number;
  nbrOfRooms!:number;
  nbrOfRoomsGroundFloor!:number;
  nbrOfParentalRooms!:number;
  nbrOfShowerRooms!:number;
  nbrOfShowerRoomsGroundFloor!:number;
  nbrOfBathrooms!:number;
  nbrOfBathroomsGroundFloor!:number;
  nbrOfToilets!:number;
  nbrOfToiletsGroundFloor!:number;
  kitchenType!:string;
  existingBasement!:boolean;
  existingWashRoom!:boolean;
  existingCave!:boolean;
  nbrOfCaves!:number;
  existingCellar!:boolean;
}
